include/sys/zio.h

```diff
@@ -304,6 +304,11 @@ typedef struct zbookmark_err_phys {
 	int64_t		zb_level;
 	uint64_t	zb_blkid;
 	uint64_t	zb_birth;
+	/*
+	 * Currrently used only when dbuf_dnode_findbp() returns EACCES when
+	 * accessing an encrypted filesystem without the key loaded.
+	 */
+	uint64_t	zb_err;
```

Comment:
We should avoid changing the on-disk format if at all possible.  Is this really needed?

Reply:
There might be a way around this, let me take another look.